vignettes/agua.Rmd

```diff
@@ -0,0 +1,294 @@
+#>  4 <split [667/76]> Fold04 <tibble [2 × 4]> <tibble [0 × 3]>
+#>  5 <split [667/76]> Fold05 <tibble [2 × 4]> <tibble [0 × 3]>
+#>  6 <split [668/75]> Fold06 <tibble [2 × 4]> <tibble [0 × 3]>
+#>  7 <split [671/72]> Fold07 <tibble [2 × 4]> <tibble [0 × 3]>
+#>  8 <split [671/72]> Fold08 <tibble [2 × 4]> <tibble [0 × 3]>
+#>  9 <split [671/72]> Fold09 <tibble [2 × 4]> <tibble [0 × 3]>
+#> 10 <split [671/72]> Fold10 <tibble [2 × 4]> <tibble [0 × 3]>
+```
+
+
+## Using H2O AutoML 
+
+Automatic machine learning (AutoML) is the process of automatically searching , screening and evaluating many models for a specific dataset. AutoML could be particularly insightful as an first-round exploratory approach to identify model families and parameterization that is most likely to succeed. You can use H2O's [AutoML](https://docs.h2o.ai/h2o/latest-stable/h2o-docs/automl.html) algorithm via the `'h2o'` engine in `auto_ml()`. Additionally, ague provides several helper functions to quickly wrangle and visualize AutoML's results. 
+
+Let's run an AutoML search on the concrete data. 
+
+```{r auto-fit, eval = FALSE}
+# run for a maximum of 120 seconds
+auto_spec <-
+  auto_ml() %>%
+  set_engine("h2o", max_runtime_secs = 120, seed = 1) %>%
+  set_mode("regression")
+
+auto_wflow <-
+  workflow() %>%
+  add_model(auto_spec) %>%
+  add_recipe(normalized_rec)
+
+auto_fit <- fit(auto_wflow, data = concrete_train)
+auto_fit %>% extract_fit_parsnip()
+#> parsnip model object
+#>
+#> ═══════════════ H2O AutoML Summary: 119 models ══════════════ 
+#> Leader Algorithm: stackedensemble
+ 
+#> Leader ID: StackedEnsemble_BestOfFamily_4_AutoML_5_20220629_160716
+#>  
+#> ══════════════════════════════════ Leaderboard ═════════════════════════════════ 
+#>                                                  model_id rmse  mse  mae rmsle
+#> 1 StackedEnsemble_BestOfFamily_4_AutoML_5_20220629_160716 4.29 18.4 2.98 0.137
+#> 2    StackedEnsemble_AllModels_2_AutoML_5_20220629_160716 4.29 18.4 2.94 0.134
+#> 3 StackedEnsemble_BestOfFamily_6_AutoML_5_20220629_160716 4.30 18.5 2.99 0.138
+#> 4            GBM_grid_1_AutoML_5_20220629_160716_model_37 4.31 18.6 3.00 0.139
+#> 5 StackedEnsemble_BestOfFamily_3_AutoML_5_20220629_160716 4.36 19.0 2.99 0.137
+#> 6    StackedEnsemble_AllModels_5_AutoML_5_20220629_160716 4.37 19.1 2.97 0.143
+#>   mean_residual_deviance
+#> 1                   18.4
+#> 2                   18.4
+#> 3                   18.5
+#> 4                   18.6
+#> 5                   19.0
+#> 6                   19.1
+```
+
+
+
+In 120 seconds, AutoML fitted 119 candidate models and returned a leaderboard of all model's cross validation performances. `model_id` is the model's identifier on the h2o server, you can retrieve any model with an id using `extract_fit_parsnip(mod, id = id)` and predict with it. By default, this extracts the best performing one. 
+
+```{r auto-predict, eval = FALSE}
+extract_fit_parsnip(auto_fit) %>%
```

Comment:
What does `predict()` when you use `predict(auto_fit)`? Should you mention that here?

Reply:
Do you mean `predict(auto_fit)` with no data or no id?